tests/components/wake_on_lan/test_switch.py

```diff
@@ -24,243 +35,288 @@ def system():
     return "Windows"
 
 
-class TestWolSwitch(unittest.TestCase):
-    """Test the wol switch."""
-
-    send_magic_packet = Mock(return_value=None)
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.hass.stop)
-
-    @patch("wakeonlan.send_magic_packet", new=send_magic_packet)
-    @patch("subprocess.call", new=call)
-    def test_valid_hostname(self):
-        """Test with valid hostname."""
-        global TEST_STATE
-        TEST_STATE = False
-        assert setup_component(
-            self.hass,
+async def test_valid_hostname(hass):
+    """Test with valid hostname."""
+    global TEST_STATE
+    TEST_STATE = False
+    assert await async_setup_component(
+        hass,
+        switch.DOMAIN,
+        {
+            "switch": {
+                "platform": "wake_on_lan",
+                "mac": "00-01-02-03-04-05",
+                "host": "validhostname",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("switch.wake_on_lan")
+    assert STATE_OFF == state.state
+
+    TEST_STATE = True
+
+    with patch.object(
+        wakeonlan, "send_magic_packet", new=send_magic_packet
+    ), patch.object(subprocess, "call", new=call):
```

Comment:
Instead of using `TEST_STATE`, you can patch `call` to return the value that you want. 

And since we always mock `send_magic_packet`, we can drop that call.

```suggestion
    with patch.object(subprocess, "call", return_value=0):
```

Reply:
I wasn't able to change all of the `send_magic_packet` patches to the fixture because three use `assert_called_with` from the Mock object. Those three I left as a Mock object patch but the others were removed and now use the `@pytest.fixture`.